util/projectRequest.ts

```diff
@@ -0,0 +1,93 @@
+import { Octokit } from '@octokit/core';
+import { paginateRest } from '@octokit/plugin-paginate-rest';
+import { Project, ACMCommitteeTopics } from './types';
+
+export async function getProjects(): Promise<Project[]> {
+  const PaginatedOctokit = Octokit.plugin(paginateRest);
+  const octokit = new PaginatedOctokit();
+  const projectsResponse = await octokit.paginate('GET /orgs/{org}/repos', {
+    org: 'uclaacm',
+    per_page: 100,
+  });
+  const sortedData = projectsResponse.sort(
+    (a, b) =>
+      new Date(b.updated_at).getTime() - new Date(a.updated_at).getTime(),
+  );
+  return sortedData.map(
+    (repo) =>
+      ({
+        name: repo.name,
+        description: repo.description,
+        link: repo.homepage || null,
+        repo: repo.html_url,
+        lang: repo.language,
+        topics: repo.topics,
+        image: getImageFromTopics(repo.topics).image,
+        alt: getImageFromTopics(repo.topics).alt,
+      } as Project),
+  ) as Project[];
```

Comment:
Are these `as Project` typecasts necessary? Are we not able to do structural typing/inference? 

(I'm not sure here, too used to using Flow at this point)

Reply:
If we define the return value of the function in the function declaration then these typecasts aren't necessary. I wasn't used to the linter requiring the function return types at the top of the function declaration so I wrote in the typecasts at first, but after running through the linter and adding in the function return types, these typecasts are redundant.

Reply:
Without the function declaration at the top or typecasts, then VS.Code's Intellisense wouldn't give the return values their types when we hover over the function, and it would not use our user defined type of the object returned.

Reply:
Right, that's what I imagined. A couple of suggestions:

* you can annotate anonymous functions in Typescript - see [More on Functions](https://www.typescriptlang.org/docs/handbook/2/functions.html)
* this also goes hand-in-hand with a way we can break down the function into constituent parts. While this isn't "mandatory" / always the right advice, we could instead do something like:
```js
function getProjectFromRepo(repo: RepoType): Project {
  return {
        name: repo.name,
        description: repo.description,
        link: repo.homepage || null,
        repo: repo.html_url,
        lang: repo.language,
        topics: repo.topics,
        image: getImageFromTopics(repo.topics).image,
        alt: getImageFromTopics(repo.topics).alt,
      };
}
```

and then pass the above function into the `map` you apply on the sort. While this is a bit more verbose, it can also make the code easier to read/understand, and you can test these functions in isolation. EOD though, up to you!